### PR TITLE
Refactor of Session.generateBoosters

### PR DESCRIPTION
**Pull request by NerdyNathan**

### CODE REVIEW

**Comment by NerdyNathan** [comment]
Reviewing with "Hide Whitespace" enabled seems to be more helpful

File: src/Session.ts [hunk 1]
```diff
@@ -887,7 +835,7 @@ export class Session implements IIndexable {
 
 	winstonTakePile() {
 		const s = this.draftState as WinstonDraftState;
-		if (!this.drafting || !s || !(s instanceof WinstonDraftState)) return false;
```
Comment (Senryoku) [inline src/Session.ts:890]: Why did you remove these checks?
If I wanted to update this, I would use type guards rather than casting the draftState unconditionally here (adding type guards for the draft states as necessary).

File: src/BoosterFactory.ts [hunk 2]
```diff
@@ -1555,13 +1567,10 @@ for (const [cid, card] of Cards) {
 
 export class PaperBoosterFactory implements IBoosterFactory {
 	set: SetInfo;
-	options: Options;
 	possibleContent: BoosterInfo[];
-	landSlot: null = null;
```
Comment (Senryoku) [inline src/BoosterFactory.ts:1560]: Didn't realize these were not used, great!

File: src/Session.ts [hunk 3]
```diff
@@ -2816,32 +2764,8 @@ export class Session implements IIndexable {
...
-	emitError(title: string = "Error", text: string | null = "Unspecified Error", showConfirmButton = true, timer = 0) {
-		Connections[this.owner]?.socket.emit("message", {
-			icon: "error",
-			title: title,
-			text: text,
-			showConfirmButton: showConfirmButton,
-			timer: timer,
-		} as Message);
-	}
-
```
Comment (Senryoku) [inline src/Session.ts:2838]: Oh yeah, I forgot to remove those!

**Review by Senryoku — COMMENTED** [review]
Looks good, thanks for the cleanup!
I can merge this and get rid of the casts/add the typeguards on draftState myself later on, unless you want to do it?

**Comment by NerdyNathan** [comment]
> Looks good, thanks for the cleanup! I can merge this and get rid of the casts/add the typeguards on draftState myself later on, unless you want to do it?
Sure, I'll swap them now

File: src/Session.ts (hunk 1 above)
Comment (NerdyNathan) [reply src/Session.ts:890]: One issue is that specifically `WinstonDraftState` fields are referenced in the rest of the method, so it needs to be cast at some point.
What's the intent here? As currently written, `!(s instanceof WinstonDraftState` will always be `false` since it was explicitly cast above. So either `s` will be `undefined` or will throw a Type Error.
